Assets/Gothic-UnZENity-Core/Scripts/ResourceLoader.cs

```diff
@@ -0,0 +1,241 @@
+            return new[] { contextPrefixPath, s }.Select(Resources.Load<GameObject>).FirstOrDefault(newPrefab => newPrefab != null);
+		});
+
+		public static void Init(string root)
+		{
+			var workPath = FindWorkPath(root);
+			var diskPaths = FindDiskPaths(root);
+
+			diskPaths.ForEach(v => Vfs.MountDisk(v, VfsOverwriteBehavior.Newer));
+			Vfs.Mount(Path.GetFullPath(workPath), "/_work", VfsOverwriteBehavior.All);
+			
+            DmLoader.AddResolver(name =>
+            {
+				var node = Vfs.Find(name);
+				return node?.Buffer.Bytes;
+            });
+		}
+
+		[CanBeNull]
+		public static ITexture TryGetTexture([NotNull] string key)
+		{
+			return Textures.TryLoad($"{GetPreparedKey(key)}-c.tex", out var item) ? item : null;
+		}
+
+		[CanBeNull]
+		public static IModelScript TryGetModelScript([NotNull] string key)
+		{
+			return ModelScript.TryLoad($"{GetPreparedKey(key)}.mds", out var item) ? item : null;
+		}
+
+		[CanBeNull]
+		public static IModelAnimation TryGetModelAnimation([NotNull] string mds, [NotNull] string key)
+		{
+			key = $"{GetPreparedKey(mds)}-{GetPreparedKey(key)}.man";
+			return ModelAnimation.TryLoad(key, out var item) ? item : null;
+		}
+
+		[CanBeNull]
+		public static IMesh TryGetMesh([NotNull] string key)
+		{
+			return Mesh.TryLoad($"{GetPreparedKey(key)}.msh", out var item) ? item : null;
+		}
+
+		[CanBeNull]
+		public static IModelHierarchy TryGetModelHierarchy([NotNull] string key)
+		{
+			return ModelHierarchy.TryLoad($"{GetPreparedKey(key)}.mdh", out var item) ? item : null;
+		}
+
+		[CanBeNull]
+		public static IModel TryGetModel([NotNull] string key)
+		{
+			return Model.TryLoad($"{GetPreparedKey(key)}.mdl", out var item) ? item : null;
+		}
+
+		[CanBeNull]
+		public static IModelMesh TryGetModelMesh([NotNull] string key)
+		{
+			return ModelMesh.TryLoad($"{GetPreparedKey(key)}.mdm", out var item) ? item : null;
+		}
+
+		[CanBeNull]
+		public static IMultiResolutionMesh TryGetMultiResolutionMesh([NotNull] string key)
+		{
+			return MultiResolutionMesh.TryLoad($"{GetPreparedKey(key)}.mrm", out var item) ? item : null;
+		}
+
+		[CanBeNull]
+		public static IMorphMesh TryGetMorphMesh([NotNull] string key)
+		{
+			return MorphMesh.TryLoad($"{GetPreparedKey(key)}.mmb", out var item) ? item : null;
+		}
+
+		[CanBeNull]
+		public static IFont TryGetFont([NotNull] string key)
+		{
+			return Font.TryLoad($"{GetPreparedKey(key)}.fnt", out var item) ? item : null;
+		}
+
+		[CanBeNull]
+		public static SoundData TryGetSound([NotNull] string key)
+		{
+			return Sound.TryLoad($"{GetPreparedKey(key)}.wav", out var item) ? item : null;
+		}
+		
+		[CanBeNull]
+		public static Segment TryGetSegment([NotNull] string key)
+		{
+			// NOTE(lmichaelis): There is no caching required here, since the loader
+			//                   already caches segments upon loading them
+			return DmLoader.GetSegment(key);
+		}
+		
+		[CanBeNull]
+		public static DaedalusVm TryGetDaedalusVm([NotNull] string key)
```

Comment:
Indeed it should not be cached because when we load a save-game for example, the VM needs to be reloaded entirely. We should manage VMs somewhere else since they need special setup anyways.

Reply:
Please add a warning, that this is a compute intensive call and point to where you cashed them as of now. Sure, you'll change, but at least highlight it for now.